http-service/src/main/java/net/runelite/http/service/hiscore/HiscoreService.java

```diff
@@ -89,6 +128,57 @@ public HiscoreResult lookup(Request request, Response response) throws IOExcepti
 		return hiscoreBuilder.build();
 	}
 
+	public SingleHiscoreSkillResult singleSkillLookup(Request request, Response response) throws IOException
+	{
+		String username = request.queryParams("username");
+		String skillname = request.params("skill");
+
+		HttpUrl hiscoreUrl = url.newBuilder()
```

Comment:
This should not be duplicated from lookup(). Split common code into a separate function.